src/time_phase.rs

```diff
@@ -0,0 +1,59 @@
+use std::io::{self, Write};
+use std::time::{Instant, Duration};
+
+
+/// RAII timer to measure how long phases take.
+// TODO does it need output parameter? or always print to stdout
+#[derive(Debug)]
+pub struct Timer<'a> {
+    output: bool,
+    name: &'a str,
+    start: Instant,
+}
+
+
+impl<'a> Timer<'a> {
+    /// Creates a Timer with the given name, and starts it.
+    ///
+    /// By default, the output stream is None, and nothing is printed
```

Comment:
I think this comment is out of date, possible referring to a time when the output was configurable.